### PR TITLE
#66 Add questions functionality

### PR DESCRIPTION
**Pull request by PublicFedorenko**
[#66 Add questions functionality](https://trello.com/c/ZdqTRSvb/66-add-questions-functionality)

### CODE REVIEW

File: frontend/src/app/shared/shared.module.ts [hunk 1]
```diff
@@ -33,5 +33,9 @@ export class SharedModule {
             'google-logo',
             sanitizer.bypassSecurityTrustResourceUrl('./assets/icons/google-logo.svg'),
         );
+        matIconRegistry.addSvgIcon(
```
Comment (apofis96) [inline frontend/src/app/shared/shared.module.ts]: Use icon from icons library

**Review by apofis96 — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/new-slot/questions/questions.component.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { IAvailabilitySlot } from '@core/models/IAvailiabilitySlot';
+import { IQuestion } from '@core/models/IQuestion';
+
+@Component({
+    selector: 'app-questions',
+    templateUrl: './questions.component.html',
+    styleUrls: ['./questions.component.sass'],
+})
+export class QuestionsComponent {
+    @Input() set newSlot(slot: IAvailabilitySlot | undefined) {
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/questions/questions.component.ts:13]:
```suggestion
    @Input() set newSlot(slot?: IAvailabilitySlot) {
```

File: frontend/src/app/modules/availability/new-slot/questions/questions.component.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        this.questions.push({ questionText: '', isMandatory: false });
+        this.onQuestionsArrayChange();
+    }
+
+    public deleteQuestion(question: IQuestion): void {
+        const index: number = this.questions.indexOf(question, 0);
+
+        if (index > -1) {
+            this.questions.splice(index, 1);
+        }
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/questions/questions.component.ts]: Please use filter instead of this

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/new-slot/questions/questions.component.ts (hunk 2 above)
Comment (PublicFedorenko) [reply frontend/src/app/modules/availability/new-slot/questions/questions.component.ts:13]: At some point I tried to use '?' but got this error and decided to move on.

A 'set' accessor cannot have an optional parameter.ts(1051) 
image

File: frontend/src/app/shared/shared.module.ts (hunk 1 above)
Comment (PublicFedorenko) [reply frontend/src/app/shared/shared.module.ts]: Done

**Review by apofis96 — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**